autopilot/contracts.go

```diff
@@ -39,3 +77,243 @@ func (ap *Autopilot) defaultContracts() ([]worker.Contract, error) {
 	}
 	return contracts, nil
 }
+
+func (ap *Autopilot) contractLoop() {
+	for {
+		select {
+		// TODO: use build var for loop interval
+		// TODO: add wakeChan (to trigger loop)
+		// TODO: run loop immediately
+		// TODO: logging
+		case <-ap.stopChan:
+			return
+		case <-time.After(time.Hour):
+		}
+
+		// don't run the contract loop if we are not synced
+		if !ap.store.Synced() {
+			continue
+		}
+
+		// fetch all contracts
+		cs, err := ap.defaultContracts()
+		if err != nil {
+			logErr(err)
+			continue
+		}
+
+		// run checks on the contract set
+		// TODO: dedupe existing contracts (?)
+		// TODO: check for IP range violations (?)
+
+		// run checks on the contracts individually
+		// TODO: separate loop (?)
```

Comment:
I'm considering pulling this out in a separate loop that marks a contract's "utility" as `!GFU` or `GFR`. I wouldn't add the term `utility` though but simply have some extra fields on the contract in the database that represent the contract utility.

Reply:
Why do you want it to be a separate loop? Since they are not really completely independent I'd just leave it in this one.